components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/handler/request/impl/DefaultLogoutRequestHandler.java

```diff
@@ -89,7 +89,7 @@ public void handle(HttpServletRequest request, HttpServletResponse response, Aut
             if (sessionContext != null) {
                 Object authenticatedUserObj = sessionContext.getProperty(FrameworkConstants.AUTHENTICATED_USER);
                 AuthenticatedUser authenticatedUser = new AuthenticatedUser();
-                if (authenticatedUserObj != null) {
+                if (authenticatedUserObj instanceof AuthenticatedUser) {
```

Comment:
Is there any possibility to have null for `authenticatedUserObj`? If so add the null check as well.